clang/include/clang/Basic/AddressSpaces.h

```diff
@@ -58,6 +58,7 @@ enum class LangAS : unsigned {
 
   // HLSL specific address spaces.
   hlsl_groupshared,
+  hlsl_private,
```

Comment:
Title and description only talks about SPIRV, but this is a purely language concept you're adding here

Reply:
That's correct, slightly reworded the description.
The thing is, only SPIR-V will be using this for now (same as the few others we'll add line hlsl_input, hlsl_output).

Reply:
> only SPIR-V will be using this for now

That's not really how languages work. It's a property of the language and exists independently of what any codegen is doing with it. As it is it sounds like you're jamming some knob you want for codegen through here that isn't part of the language.

This needs reference to some HLSL spec describing the address space without any mention of SPIRV.

Reply:
This being a small stepping stone for the larger change bringing hlsl_input/hlsl_output, I haven't wrote a specific HLSL spec part, only relied on current MSDN & HLSL implementations (I agree, not great).
For the `hlsl_input`/`hlsl_output` address spaces, here is the wg-hlsl proposal:
https://github.com/llvm/wg-hlsl/pull/97


But you are right, maybe that's something we should now write properly in the hlsl-spec repo